roles/metrics-export.lua

```diff
@@ -339,23 +339,49 @@ local function apply_http(conf)
                     error("failed to parse URI: " .. err, 2)
                 end
                 target = {
-                    value = node.listen,
-                    is_httpd_role = false,
+                    value = 'listen_' .. host .. ':' .. tostring(port),
                 }
             else
                 target = {
-                    value = httpd_role.DEFAULT_SERVER_NAME,
-                    is_httpd_role = true,
+                    value = 'httpd_' .. httpd_role.DEFAULT_SERVER_NAME,
+                    httpd_name = httpd_role.DEFAULT_SERVER_NAME,
                 }
             end
 
             http_servers = http_servers or {}
+
             -- Since the 'listen' and 'server' names of other servers in the config may be
-            -- the same, we create a unique string concatenating the key name and information
-            -- about whether it is an httpd key or not.
-            enabled[tostring(target.value) .. tostring(target.is_httpd_role)] = true
+            -- the same, we create a unique string concatenating the key (which is listen or server)
+            -- and its value.
+            enabled[target.value] = true
+
+            if http_servers[target.value] == nil then
+                if target.httpd_name == nil then
+                    local count_global_addrs = 0
+
+                    for _, n in pairs(conf) do
+                        -- Count if it is more than one server that on gloabl address 0.0.0.0.
+                        if n.listen ~= nil and string.find(tostring(n.listen), tostring(port)) ~= nil then
+                            count_global_addrs = count_global_addrs + 1
+                        end
+                    end
+
+                    if host == '0.0.0.0' then
+                        if count_global_addrs == 1 then
+                            for k in pairs(http_servers) do
+                                if http_servers[k].httpd_name == nil and string.find(k, ':' .. port) ~= nil then
+                                    http_servers[k].httpd:stop()
+                                    enabled[k] = false
+                                end
+                            end
+                        end
+                    elseif http_servers['listen_0.0.0.0:' .. port] ~= nil and count_global_addrs == 1 then
+                        local key = 'listen_0.0.0.0:' .. port
+                        http_servers[key].httpd:stop()
```

Comment:
On the other hand, we should think whether roles using httpd role servers (like tcf ones) are prepared for server reload and make some changes (together with documentation tips here), if not.